langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiEmbeddingModel.java

```diff
@@ -100,17 +128,20 @@ private Response<List<Embedding>> embedTexts(List<String> texts) {
 
             List<String> batch = texts.subList(i, Math.min(i + BATCH_SIZE, texts.size()));
 
-            EmbeddingRequest request = EmbeddingRequest.builder()
-                    .input(batch)
-                    .build();
-
-            EmbeddingResponse response = withRetry(() -> client.embedding(request).execute(), maxRetries);
+            EmbeddingsOptions options = new EmbeddingsOptions(batch);
+            Embeddings response =  withRetry(() -> client.getEmbeddings(deploymentName, options), maxRetries);
 
-            embeddings.addAll(response.data().stream()
-                    .map(openAiEmbedding -> Embedding.from(openAiEmbedding.embedding()))
-                    .collect(toList()));
+            for (EmbeddingItem embeddingItem : response.getData()) {
+                List<Double> openAiVector = embeddingItem.getEmbedding();
+                float[] langChainVector = new float[openAiVector.size()];
+                for (int index = 0; index < openAiVector.size(); index++) {
+                    langChainVector[index] = openAiVector.get(index).floatValue();
```

Comment:
optional: would be nice to move this logic into `public static Embedding from(List<Double> vector) {...}` inside `Embedding`

Reply:
Done